src/Microsoft.Health.Fhir.Shared.Web/Startup.cs

```diff
@@ -57,17 +61,42 @@ public virtual void ConfigureServices(IServiceCollection services)
                 });
             }
 
+            PrometheusMetricsConfig prometheusConfig = new PrometheusMetricsConfig();
+            Configuration.Bind("PrometheusMetrics", prometheusConfig);
+            services.AddSingleton(Options.Create(prometheusConfig));
+
+            if (prometheusConfig.Enabled)
+            {
+                services.Add<PrometheusMetricsServer>()
+                    .Singleton()
+                    .AsService<IStartable>();
+
+                if (prometheusConfig.SystemMetrics)
+                {
+                    services.AddSystemMetrics();
+                }
+            }
+
             AddApplicationInsightsTelemetry(services);
         }
 
         // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
-        public virtual void Configure(IApplicationBuilder app)
+        public virtual void Configure(IApplicationBuilder app, IOptions<PrometheusMetricsConfig> prometheusConfig)
         {
             if (string.Equals(Configuration["ASPNETCORE_FORWARDEDHEADERS_ENABLED"], "true", StringComparison.OrdinalIgnoreCase))
             {
                 app.UseForwardedHeaders();
             }
 
+            var prometheusMetricsConfig = prometheusConfig.Value;
+            if (prometheusMetricsConfig.Enabled)
+            {
+                if (prometheusMetricsConfig.HttpMetrics)
+                {
+                    app.UseHttpMetrics();
+                }
+            }
```

Comment:
Perhaps to make this cleaner, we could follow `app.UsePrometheusMetrics()` pattern.
Then resolve the config with `app.ApplicationServices.GetService<>` instead of changing this Configure signature